src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -423,30 +710,137 @@ private static LevelsClause ParseLevels(long? levelsOption, IEdmType sourceType,
+            }
+        }
+
+        private static bool VerifySelectedNavigationProperty(IList<ODataPathSegment> selectedPath, SelectTermToken tokenIn)
+        {
+            NavigationPropertySegment navPropSegment = selectedPath.LastOrDefault() as NavigationPropertySegment;
+            if (navPropSegment != null)
+            {
+                // After navigation property, it's not allowed to nest query options
+                VerifyNoQueryOptionsNested(tokenIn, navPropSegment.Identifier);
+
+                return true;
+            }
+
+            return false;
+        }
+
+        private static void VerifyNoQueryOptionsNested(SelectTermToken selectToken, string identifier)
+        {
+            if (selectToken != null)
+            {
+                if (selectToken.ComputeOption != null ||
+                    selectToken.FilterOption != null ||
+                    selectToken.OrderByOptions != null ||
+                    selectToken.SearchOption != null ||
+                    selectToken.CountQueryOption != null ||
+                    selectToken.SelectOption != null ||
+                    selectToken.TopOption != null ||
+                    selectToken.SkipOption != null)
+                {
+                    throw new ODataException(ODataErrorStrings.SelectExpandBinder_InvalidQueryOptionNestedSelection(identifier));
+                }
+            }
+        }
+
+        internal static void AddToSelectedItems(SelectItem itemToAdd, List<SelectItem> selectItems)
+        {
+            if (itemToAdd == null)
+            {
+                return;
+            }
+
+            // ignore all property selection if there's a wildcard select item.
+            if (selectItems.Any(x => x is WildcardSelectItem) && IsStructuralOrNavigationPropertySelectionItem(itemToAdd))
+            {
+                return;
+            }
+
+            // if the selected item is a nav prop, then see if its already there before we add it.
+            PathSelectItem pathSelectItem = itemToAdd as PathSelectItem;
+            if (pathSelectItem != null)
+            {
+                NavigationPropertySegment trailingNavPropSegment = pathSelectItem.SelectedPath.LastSegment as NavigationPropertySegment;
+                if (trailingNavPropSegment != null)
+                {
+                    if (selectItems.OfType<PathSelectItem>().Any(i => i.SelectedPath.Equals(pathSelectItem.SelectedPath)))
+                    {
+                        return;
+                    }
+                }
+            }
+
+            // if the selected item is "*", filter the existing property selection.
+            if (itemToAdd is WildcardSelectItem)
+            {
+                var shouldFilter = selectItems.Where(s => IsStructuralSelectionItem(s)).ToList();
+                foreach (var filterItem in shouldFilter)
+                {
+                    selectItems.Remove(filterItem);
+                }
+            }

```

Comment:
Alternatively, iterate through the selectItems backward (by index) and remove any items for which IsStructuralSelectionItem is true:

            // iterate the list back to front, removing structural selection items
            for (int i = selectItems.Count -1; i >= 0; i--)
            {
                if(IsStructuralSelectionItem(selectItems[i]))
                {
                    selectItems.RemoveAt(i);
                }
            }

This avoids the LINQ query, the extra list, and having to go back and look up each item by value to remove it.  Probably not a big perf issue for this case, but avoiding LINQ is good practice in general.